components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/internal/util/AuthenticatedUserBuilder.java

```diff
@@ -179,22 +181,60 @@ private User resolveLocalUserFromUserStore(String userId, AuthenticatedUserData.
         throw new ActionExecutionResponseProcessorException(errorMessage);
     }
 
-    private Map<ClaimMapping, String> resolveUserNameAndClaimsFromResponse() {
+    private Map<ClaimMapping, String> resolveUserNameAndClaimsFromResponse()
+            throws ActionExecutionResponseProcessorException {
 
         Map<ClaimMapping, String> userAttributes = new HashMap<>();
         for (AuthenticatedUserData.Claim claim : userFromResponse.getUser().getClaims()) {
+            validateClaimValue(claim.getUri(), claim.getValue());
             if (GROUP_CLAIM.equals(claim.getUri())) {
                 ignoreGroupsInClaimsInResponse();
                 continue;
+            } else if (ROLES_CLAIM.equals(claim.getUri())) {
+                DiagnosticLogger.logSuccessResponseDataValidationError(new AuthenticationActionExecutionResult(
+                        "claims/" + ROLES_CLAIM, Availability.AVAILABLE, Validity.INVALID,
+                        "Currently we are not supporting role management with the custom authenticator, " +
```

Comment:
Let's simplify.

"Currently setting roles for the authenticated user is not allowed." Check if there are better options